kubespawner/reflector.py

```diff
@@ -150,12 +139,16 @@ def _list_and_update(self):
 
         Overwrites all current resource info.
         """
-        initial_resources = getattr(self.api, self.list_method_name)(
-            self.namespace,
-            label_selector=self.label_selector,
-            field_selector=self.field_selector,
-            _request_timeout=self.request_timeout,
-        )
+        list_args = {
+            'label_selector': self.label_selector,
```

Comment:
This should include `self.namespace` if self.namespace, right?

Reply:
see comments below about different implementation approaches.